src/Symfony/Component/HttpFoundation/StreamedJsonResponse.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpFoundation;
+
+use function array_keys;
+use function explode;
+use function flush;
+
+use Generator;
+
+use function json_encode;
+
+use const JSON_THROW_ON_ERROR;
+
+/**
+ * StreamedJsonResponse represents a streamed HTTP response for JSON.
+ *
+ * A StreamedJsonResponse uses a structure and generics to create an
+ * efficient resource saving JSON response.
+ *
+ * It uses flush after a specified flush size to directly stream the data.
+ *
+ * @see flush()
+ *
+ * @author Alexander Schranz <alexander@sulu.io>
+ *
+ * Example usage:
+ *
+ * $response = new StreamedJsonResponse(
+ *     // json structure with replace identifiers
+ *     [
+ *         '_embedded' => [
+ *             'articles' => '__articles__',
+ *         ],
+ *     ],
+ *     // array of generators with replace identifier used as key
+ *     [
+ *         '__articles__' => function (): \Generator { // any method or function returning a Generator
+ *              yield ['title' => 'Article 1'];
+ *              yield ['title' => 'Article 2'];
+ *              yield ['title' => 'Article 3'];
+ *         },
+ *     ]
+ * );
+ */
+class StreamedJsonResponse extends StreamedResponse
+{
+    public const DEFAULT_ENCODING_OPTIONS = JsonResponse::DEFAULT_ENCODING_OPTIONS;
+
+    protected int $encodingOptions = self::DEFAULT_ENCODING_OPTIONS;
+
+    /**
+     * @param mixed[] $structure
+     * @param array<string, Generator<int|string, mixed> $generics
```

Comment:
Why suggesting `int` as key if it's used as separator in `explode`? Does not look safe to do so.